After close symfony:serve, can't run it correctly

After close symfony:serve, can't run it correctly.

But if change the PHP version on .php-version or delete the ~/.symfony5/var/xxxx/xxxx.pid it's running well

I've got the same problem, so I'll try to add some more info to this:

* Symfony CLI version 5.0.7 (c) 2017-2022 Symfony SAS (2022-01-03T09:39:22Z - stable)
* PHP 8.0.13 (cli) (built: Nov 28 2021 13:11:14) ( NTS )
* macOS 12.1 @ M1
* The CLI works once, but consecutive starts fail to launch php-fpm - the only output is an info that the server is tailing two log files and then the process just hangs there
* The only error I can see is `{"level":"error","message":"Unable to start PHP-FPM"}` when I hit `^C`
* Removing the PID file (like @gbere) said solves the problem, once - the PID file must be removed every time in order to start the server

I'm not sure if this is relevant, but:

* I had no issues with the manually installed v4 version. I removed the file and used homebrew to install v5, which is where the problem started
* Every time I start the binary, my firewall is asking me whether I want to allow incoming connections. This has not been happening with the previous version. Most likely irrelevant, though.

Here is my entire console command capturing a failed start, removal of the PID file, and consecutive restart. 

```
janklan@pb app % symfony serve
Tailing Web Server log file (/Users/janklan/.symfony5/log/7240ac7dcb5ca7abcc6cecd516f3f01a5ceca0ca.log)
Tailing PHP-FPM log file (/Users/janklan/.symfony5/log/7240ac7dcb5ca7abcc6cecd516f3f01a5ceca0ca/53fb8ec204547646acb3461995e4da5a54cc7575.log)
^C{"level":"error","message":"Unable to start PHP-FPM"}
                                                                                                                        
 [ERROR] PHP-FPM failed to start:                                                                                       
 fpm is running, pid 2648                                                                                               
 ready to handle connections                                                                                            
 Terminating ...                                                                                                        
                                                                                                                        
janklan@pb app % rm ~/.symfony5/var/7240ac7dcb5ca7abcc6cecd516f3f01a5ceca0ca/eae5a91f67e6ab516cf766bd2eebd5e4b5dfd1be.pid
janklan@pb app % symfony serve                                                                                           
Tailing Web Server log file (/Users/janklan/.symfony5/log/7240ac7dcb5ca7abcc6cecd516f3f01a5ceca0ca.log)
Tailing PHP-FPM log file (/Users/janklan/.symfony5/log/7240ac7dcb5ca7abcc6cecd516f3f01a5ceca0ca/53fb8ec204547646acb3461995e4da5a54cc7575.log)
                                                                                                                        
 [OK] Web server listening                                                                                              
      The Web server is using PHP FPM 8.0.13                                                                            
      https://127.0.0.1:8000                                                                                            
                                                                                                                        

[Web Server ] Jan  4 10:20:17 |DEBUG  | PHP    Reloading PHP versions 
[Web Server ] Jan  4 10:20:17 |DEBUG  | PHP    Using PHP version 8.0.13 (from .php-version from current dir: /Users/janklan/dev/foo/.php-version) 
```

--- a/commands/local_server_start.go
+++ b/commands/local_server_start.go
@@ -89,6 +89,15 @@ var localServerStartCmd = &console.Command{
 
 		homeDir := util.GetHomeDir()
 
+		shutdownCh := make(chan bool, 1)
+		go func() {
+			sigsCh := make(chan os.Signal, 1)
+			signal.Notify(sigsCh, syscall.SIGINT, syscall.SIGQUIT, syscall.SIGTERM)
+			<-sigsCh
+			signal.Stop(sigsCh)
+			shutdownCh <- true
+		}()
+
 		if c.Bool("daemon") && !reexec.IsChild() {
 			varDir := filepath.Join(homeDir, "var")
 			if err := os.MkdirAll(varDir, 0755); err != nil {
@@ -328,14 +337,6 @@ var localServerStartCmd = &console.Command{
 		if reexec.IsChild() {
 			terminal.RemapOutput(lw, lw).SetDecorated(true)
 		}
-		shutdownCh := make(chan bool, 1)
-		go func() {
-			sigsCh := make(chan os.Signal, 1)
-			signal.Notify(sigsCh, syscall.SIGINT, syscall.SIGQUIT, syscall.SIGTERM)
-			<-sigsCh
-			signal.Stop(sigsCh)
-			shutdownCh <- true
-		}()
 
 		select {
 		case err := <-errChan:

--- a/commands/local_server_start.go
+++ b/commands/local_server_start.go
@@ -30,7 +30,6 @@ import (
 	"os/signal"
 	"path/filepath"
 	"syscall"
-	"time"
 
 	"github.com/pkg/errors"
 	"github.com/rs/zerolog"
@@ -47,6 +46,7 @@ import (
 	"github.com/symfony-cli/symfony-cli/reexec"
 	"github.com/symfony-cli/symfony-cli/util"
 	"github.com/symfony-cli/terminal"
+	"golang.org/x/sync/errgroup"
 )
 
 var localServerStartCmd = &console.Command{
@@ -275,10 +275,6 @@ var localServerStartCmd = &console.Command{
 			if err := pidFile.Write(os.Getpid(), port, scheme); err != nil {
 				return err
 			}
-			defer func() {
-				pidFile.Remove()
-				terminal.Eprintln("Shut down, bye!")
-			}()
 
 			reexec.NotifyForeground("listening")
 			ui.Success(msg)
@@ -345,9 +341,31 @@ var localServerStartCmd = &console.Command{
 		case err := <-errChan:
 			return err
 		case <-shutdownCh:
-			ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
-			defer cancel()
-			p.HTTP.Stop(ctx)
+			terminal.Eprintln("")
+			terminal.Eprintln("Shutting down!")
+			pids := pid.AllWorkers(projectDir)
+			var g errgroup.Group
+			running := 0
+			for _, p := range pids {
+				terminal.Eprintf("Stopping <comment>%s</>", p.ShortName())
+				if p.IsRunning() {
+					running++
+					g.Go(p.Stop)
+					terminal.Eprintln("")
+				} else {
+					terminal.Eprintln(": <comment>not running</>")
+				}
+			}
+			if err := g.Wait(); err != nil {
+				return err
+			}
+			terminal.Eprintln("Stopping <comment>Web Server</>")
+			if err := pidFile.Remove(); err != nil {
+				running++
+				return err
+			}
+			terminal.Eprintln("")
+			ui.Success(fmt.Sprintf("Stopped %d process(es) successfully", running))
 		}
 		return nil
 	},

--- a/commands/local_server_start.go
+++ b/commands/local_server_start.go
@@ -86,6 +86,9 @@ var localServerStartCmd = &console.Command{
 			ui.Warning("The local web server is already running")
 			return errors.WithStack(printWebServerStatus(projectDir))
 		}
+		if err := cleanupWebServerFiles(projectDir, pidFile); err != nil {
+			return err
+		}
 
 		homeDir := util.GetHomeDir()
 
@@ -344,30 +347,29 @@ var localServerStartCmd = &console.Command{
 		case <-shutdownCh:
 			terminal.Eprintln("")
 			terminal.Eprintln("Shutting down!")
-			pids := pid.AllWorkers(projectDir)
-			var g errgroup.Group
-			running := 0
-			for _, p := range pids {
-				terminal.Eprintf("Stopping <comment>%s</>", p.ShortName())
-				if p.IsRunning() {
-					running++
-					g.Go(p.Stop)
-					terminal.Eprintln("")
-				} else {
-					terminal.Eprintln(": <comment>not running</>")
-				}
-			}
-			if err := g.Wait(); err != nil {
-				return err
-			}
-			terminal.Eprintln("Stopping <comment>Web Server</>")
-			if err := pidFile.Remove(); err != nil {
-				running++
+			if err := cleanupWebServerFiles(projectDir, pidFile); err != nil {
 				return err
 			}
 			terminal.Eprintln("")
-			ui.Success(fmt.Sprintf("Stopped %d process(es) successfully", running))
+			ui.Success("Stopped all processes successfully")
 		}
 		return nil
 	},
+}
+
+func cleanupWebServerFiles(projectDir string, pidFile *pid.PidFile) error {
+	pids := pid.AllWorkers(projectDir)
+	var g errgroup.Group
+	for _, p := range pids {
+		if p.IsRunning() {
+			g.Go(p.Stop)
+		}
+	}
+	if err := g.Wait(); err != nil {
+		return err
+	}
+	if err := pidFile.Remove(); err != nil {
+		return err
+	}
+	return nil
 }

--- a/local/http/http.go
+++ b/local/http/http.go
@@ -20,7 +20,6 @@
 package http
 
 import (
-	"context"
 	"crypto/tls"
 	"fmt"
 	"net"
@@ -37,7 +36,6 @@ import (
 	"github.com/symfony-cli/cert"
 	"github.com/symfony-cli/symfony-cli/local/html"
 	"github.com/symfony-cli/symfony-cli/local/process"
-	"golang.org/x/sync/errgroup"
 )
 
 // ServerCallback serves non-static HTTP resources
@@ -123,24 +121,6 @@ func (s *Server) Start(errChan chan error) (int, error) {
 	}()
 
 	return port, nil
-}
-
-// Stop stops the server
-func (s *Server) Stop(ctx context.Context) error {
-	var g errgroup.Group
-	g.Go(func() error {
-		if s.httpsserver != nil {
-			return errors.WithStack(s.httpsserver.Shutdown(ctx))
-		}
-		return nil
-	})
-	g.Go(func() error {
-		if s.httpserver != nil {
-			return errors.WithStack(s.httpserver.Shutdown(ctx))
-		}
-		return nil
-	})
-	return g.Wait()
 }
 
 // ProxyHandler wraps the regular handler to log the HTTP messages

